src/utils/database_util.py

```diff
@@ -5,44 +5,60 @@
 from typing import List
 from tinydb import TinyDB, where
 
+
 class DatabaseUtil:
     """
-        A class for managing all database interactions
+    A class for managing all database interactions
     """
 
     TABLE_LEETCODE_QUESTION = "Leetcode_Question"
+    TABLE_LEETCODE_USER = "Leetcode_User"
 
     def __init__(self, database_path):
         self.database_path = database_path
         # Once we migrate to Docker,
         #  put this in a Docker volume such as '/leetcode_data/db.json'
         self.db = TinyDB(self.database_path)
 
-    def table_leetcodequestion_insert_many(self, items : List[dict]) -> int:
+    def table_leetcodequestion_insert_many(self, items: List[dict]) -> int:
         """
-            Inserts a collection of items into the Leetcode_Question database
-            table
+        Inserts a collection of items into the Leetcode_Question database
+        table
         """
+
         table = self.db.table(self.TABLE_LEETCODE_QUESTION)
         inserts = 0
         for item in items:
-            matches = table.search(where('id') == item['id'])
+            matches = table.search(where("id") == item["id"])
             if len(matches) == 0:
                 table.insert(item)
                 inserts += 1
         return inserts
 
     def table_leetcodequestion_loadall(self):
         """
-            Loads all items in the Leetcode_Question database table
+        Loads all items in the Leetcode_Question database table
         """
         table = self.db.table(self.TABLE_LEETCODE_QUESTION)
         return table.all()
 
-    def table_leetcodequestion_delete(self, question_id : int):
+    def table_leetcodequestion_delete(self, question_id: int):
         """
-            Deletes an item in the Leetcode_Question database table
+        Deletes an item in the Leetcode_Question database table
         """
 
         table = self.db.table(self.TABLE_LEETCODE_QUESTION)
         return table.remove(where("id") == question_id)
+
+    def table_leetcodeuser_insert(self, items: List[dict]) -> int:
+        """
+        Inserts a collection of items in the LeetcodeUser database table
+        """
+        table = self.db.table(self.TABLE_LEETCODE_USER)
+        inserts = 0
+        for item in items:
+            matches = table.search(where("id") == item["id"])
+            if len(matches) == 0:
+                table.insert(item)
+                inserts += 1
+        return inserts
```

Comment:
I should have paid closer attention to this before, but this code is actually problematic.

1) The idea is taken from the `table_leetcodequestion_insert_many` method, which is designed for inserting a collection of questions. However, we don't have the same need for inserting users in bulk, as we will most likely be inserting users individually. Therefore, if we are adding a `_insert_many` method for this table, it should be named accordingly
2) The Leetcode_User object does not have an "id" attribute per the schema. It should have two fields, "discord_id" and "leetcode_id". The point of this table is to associate Discord ID's with Leetcode ID's so we can correctly lookup a Discord User's leetcode problem statuses. Thus, this method should check to see if the item to be inserted has a `leetcode_id` or `discord_id` that is already associated with an existing user. 

All of that said, I discussed this with BaiChi recently who is working on the `_leetcodeuser_load` methods and he should be preparing an update for this `_insert` method.